### ISSUE TITLE
Create a CRUD example

### ISSUE THREAD

**Issue by jhollingworth** [jhollingworth/marty#81]

**Comment by jonhester** [comment]
I would definitely love to see this.

**Comment by nhagen** [comment]
My question would be covered by this, but I'm having trouble imagining how Create works.

```Javascript
var userStore = Marty.createStore({
...
  createUser: function(user) {
    return this.fetch({
      id: 'CREATE_USER',
      locally: function() {
        return this.state.users[id]; // We don't know id
      },
      remotely: function() {
        return userApi.createUser(user);
      }
    });
  }
...
});
```

If you don't know id, how can local ever return something other than undefined?

**Comment by AlanVerbner** [comment]
+1

**Comment by idolizesc** [comment]
Yes, this would be hugely helpful. The Todo example is outdated and intentionally avoids using a lot of the more important features of Marty (fetch API, promises, state sources, etc)

**Comment by jhollingworth** [comment]
Just as an FYI there is a [chat example](https://github.com/martyjs/marty-chat-example/) ([es6 version](https://github.com/martyjs/marty-chat-example-es6/)) which is bit more advanced

**Comment by AlanVerbner** [comment]
The problems I faced with a CRUD (most of apps I need to work with have them :cry:)  are:
- Creating an item (which does not have an id which might be created by db)
- Always query to the server for data on the grids (this is related with https://github.com/martyjs/marty/issues/111)
- Adding an item while you have a paginated and sorted grid (do you manage sorting clientside?). If you query to the server without caching it, this won't be a problem.

I haven't found a direct match against those items on the examples. Reading the issues alongside with examples helped me to find my own way to implement a CRUD, though.
